File: seed.js
import fs from "fs";

const START = 1;
const END = 493;
const DELAY = 300;

const sleep = (ms) => new Promise((r) => setTimeout(r, ms));

async function fetchPokemon(id) {
  const url = `https://pokeapi.co/api/v2/pokemon/${id}`;

  try {
    const res = await fetch(url);

    if (!res.ok) {
      throw new Error(`HTTP ${res.status}`);
    }

    return await res.json();
  } catch (err) {
    console.error(`Failed for #${id}:`, err.message);
    return null;
  }
}

async function main() {
  const results = [];

  console.log("Seeding Gen 1–4 via Node...");

  for (let id = START; id <= END; id++) {
    const data = await fetchPokemon(id);

    if (!data) continue;

    const generation =
      id <= 151 ? 1 :
      id <= 251 ? 2 :
      id <= 386 ? 3 : 4;

    results.push({
      id: data.id,
      name: data.name,
      generation,
      sprite_url:
        data.sprites.other["official-artwork"].front_default ||
        data.sprites.front_default
    });

    console.log(`Fetched ${data.name}`);
    await sleep(DELAY);
  }

  fs.writeFileSync(
    "pokemon_gen1_4.json",
    JSON.stringify(results, null, 2)
  );

  console.log("Done. Saved to pokemon_gen1_4.json");
}

main();

File: data/seed.py
import requests
import time
import json

START = 1
END = 493
DELAY = 0.3
MAX_RETRIES = 3

pokemon_data = []

print("Seeding Gen 1–4 Pokémon into JSON...")

for poke_id in range(START, END + 1):
    url = f"https://pokeapi.co/api/v2/pokemon/{poke_id}"
    attempt = 0
    data = None

    while attempt < MAX_RETRIES:
        try:
            response = requests.get(url, timeout=10)
            response.raise_for_status()
            data = response.json()
            break
        except Exception as e:
            attempt += 1
            print(f"Retry {attempt} for #{poke_id}...")
            time.sleep(1)

    if not data:
        print(f"Failed permanently: #{poke_id}")
        continue

    generation = (
        1 if poke_id <= 151 else
        2 if poke_id <= 251 else
        3 if poke_id <= 386 else
        4
    )

    pokemon_data.append({
        "id": data["id"],
        "name": data["name"],
        "generation": generation,
        "sprite_url": (
            data["sprites"]["other"]["official-artwork"]["front_default"]
            or data["sprites"]["front_default"]
        )
    })

    print(f"Fetched {data['name']} (Gen {generation})")

    time.sleep(DELAY)

with open("pokemon_gen1_4.json", "w") as f:
    json.dump(pokemon_data, f, indent=2)

print("Done. Data saved to pokemon_gen1_4.json")

